dev/Microsoft.Process.Environment/EnvironmentManager.cpp

```diff
@@ -63,4 +74,125 @@ namespace winrt::Microsoft::ProjectReunion::implementation
+        wil::unique_hkey environmentVariablesHKey = GetRegHKeyForEVUserAndMachineScope();
+
+        // While this way of calculating the max size of the names,
+        // values, and total number of entries includes two calls
+        // to the registry, I believe this is superior to
+        // using a do/while or a while with a prime
+        // because there is no chance of the loop iterating more than
+        // is needed AND the size of the name and value arrays are
+        // only as big as the biggest name or value.
+
+        DWORD sizeOfLongestNameInCharacters;
+        DWORD sizeOfLongestValueInCharacters;
+        DWORD numberOfValues;
+
+        THROW_IF_FAILED(RegQueryInfoKeyW(
+            environmentVariablesHKey.get()
+            , nullptr // lpClass
+            , nullptr // lpcchClass
+            , nullptr // lpReserved
+            , nullptr // lpcSubKeys
+            , nullptr // lpcbMaxSubKeyLen
+            , nullptr // lpcbMaxCLassLen
+            , &numberOfValues //lpcValues
+            , &sizeOfLongestNameInCharacters
+            , &sizeOfLongestValueInCharacters
+            , nullptr // lpcmSecurityDescriptor
+            , nullptr)); // lpftLastWriteTime
+
+        // +1 for null character
+        const DWORD NAME_LENGTH = sizeOfLongestNameInCharacters + 1;
+        const DWORD VALUE_SIZE_IN_BYTES = sizeOfLongestValueInCharacters * sizeof(WCHAR);
+
+        std::unique_ptr<wchar_t[]> environmentVariableName(new wchar_t[NAME_LENGTH]);
+        std::unique_ptr<BYTE[]> environmentVariableValue(new BYTE[VALUE_SIZE_IN_BYTES]);
+
+        for (int valueIndex = 0; valueIndex < numberOfValues; valueIndex++)
+        {
+            DWORD nameLength = NAME_LENGTH;
+            DWORD valueSize = VALUE_SIZE_IN_BYTES;
+            LSTATUS enumerationStatus = RegEnumValueW(environmentVariablesHKey.get()
+                , valueIndex
+                , environmentVariableName.get()
+                , &nameLength
+                , nullptr
+                , nullptr
+                , environmentVariableValue.get()
+                , &valueSize);
+
+            // An empty name indicates the default value.
+            if (nameLength == 0)
+            {
+                continue;
+            }
+
+            // If there was an error getting the value
+            if (enumerationStatus != ERROR_SUCCESS && enumerationStatus != ERROR_NO_MORE_ITEMS)
+            {
+                THROW_HR(HRESULT_FROM_WIN32(enumerationStatus));
+            }
+
+            environmentVariables.Insert(environmentVariableName.get(), reinterpret_cast<LPWSTR>(environmentVariableValue.get()));
+
+            environmentVariableName.reset(new wchar_t[NAME_LENGTH]);
+            environmentVariableValue.reset(new BYTE[VALUE_SIZE_IN_BYTES]);
+        }
+
+        return environmentVariables;
+    }
+
+    wil::unique_hkey EnvironmentManager::GetRegHKeyForEVUserAndMachineScope(bool needsWriteAccess)
+    {
+        assert(m_Scope != Scope::Process);
```

Comment:
Don't bother with assert. Same lesson as Windows CHK builds -- unless you routinely use them you get no value (except surprises when they break). And we don't use them.

Alternatives:
* Do nothing
* Error handling e.g. RETURN_HR_IF(E_INVALIDARG, ...)
* FailFast e.g. FAIL_FAST_HR_IF(E_INVALIDARG, ...)